frontend/pkg/frontend/otel_sdk.go

```diff
@@ -0,0 +1,77 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// TODO: can we move this into ocm-sdk-go/tracing?
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/contrib/exporters/autoexport"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+func InstallOpenTelemetryTracer(ctx context.Context, logger *slog.Logger, resourceAttrs ...attribute.KeyValue) (
+	func(context.Context) error,
+	error,
+) {
+	// NOTE: Auto span exporter sends traces to https://localhost:4318/v1/traces by default.
+	// We overwrite the default value with "none".
+	// See:
+	// https://github.com/open-telemetry/opentelemetry-go-contrib/blob/f6667f6f9eab2370f46d0903cf323cda3b7ca2bd/exporters/autoexport/spans.go#L32-L60
+	if v, _ := os.LookupEnv("OTEL_TRACES_EXPORTER"); v == "" {
```

Comment:
(nit) https://pkg.go.dev/go.opentelemetry.io/contrib/exporters/autoexport#WithFallbackSpanExporter seems to imply that we can override the default exporter without messing up the env vars.